src/Boost.sol

```diff
@@ -1,87 +1,145 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.4;
 
-import "openzeppelin-contracts/token/ERC20/IERC20.sol";
+import "openzeppelin-contracts/access/Ownable.sol";
 import "openzeppelin-contracts/utils/cryptography/SignatureChecker.sol";
 import "openzeppelin-contracts/utils/cryptography/draft-EIP712.sol";
+
 import "./IBoost.sol";
 
-contract Boost is IBoost, EIP712("boost", "1") {
+contract Boost is IBoost, EIP712("boost", "1"), Ownable {
     bytes32 public immutable eip712ClaimStructHash =
         keccak256("Claim(uint256 boostId,address recipient,uint256 amount)");
 
     uint256 public nextBoostId = 1;
     mapping(uint256 => BoostConfig) public boosts;
     mapping(address => mapping(uint256 => bool)) public claimed;
 
+    // Constant eth fee (in gwei) that is the same for all boost creators.
+    uint256 public ethFee;
+    // The fraction of the total boost deposit that is taken as a fee.
+    // represented as an integer denominator (1/x)%
+    uint256 public tokenFee;
+
+    constructor(address _protocolOwner, uint256 _ethFee, uint256 _tokenFee) {
+        setEthFee(_ethFee);
+        setTokenFee(_tokenFee);
+        transferOwnership(_protocolOwner);
+    }
+
+    function setEthFee(uint256 _ethFee) public override onlyOwner {
+        ethFee = _ethFee;
+        emit EthFeeSet(_ethFee);
+    }
+
+    function setTokenFee(uint256 _tokenFee) public override onlyOwner {
+        tokenFee = _tokenFee;
+        emit TokenFeeSet(_tokenFee);
+    }
+
+    function collectEthFees() external onlyOwner {
+        payable(msg.sender).transfer(address(this).balance);
+    }
+
     /// @notice Create a new boost and transfer tokens to it
-    function createBoost(BoostConfig calldata boost) external override {
-        if (boost.balance == 0) revert BoostDepositRequired();
-        if (boost.end <= block.timestamp) revert BoostEndDateInPast();
-        if (boost.start >= boost.end) revert BoostEndDateBeforeStart();
-        if (boost.guard == address(0)) revert InvalidGuard();
+    function createBoost(
+        string calldata _strategyURI,
+        IERC20 _token,
+        uint256 _amount,
+        address _guard,
+        uint256 _start,
+        uint256 _end,
+        address _owner
+    ) external payable override {
+        if (_amount == 0) revert BoostDepositRequired();
+        if (_end <= block.timestamp) revert BoostEndDateInPast();
+        if (_start >= _end) revert BoostEndDateBeforeStart();
+        if (_guard == address(0)) revert InvalidGuard();
+        if (msg.value < ethFee) revert InsufficientEthFee();
+
+        uint256 balance = 0;
+        if (tokenFee > 0) {
+            uint256 tokenFeeAmount = _amount / tokenFee;
```

Comment:
This brings the question: should we also disallow `tokenFee = 1`? Creating a boost with `0` makes no sense I think?

Reply:
the protocol owner sets the fee so they can just not set it to 1. do you think we should actually prevent it?

Reply:
yeah just wondering why would anyone want to use boost if `tokenFee == 1`